hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/ExtendedInputStream.java

```diff
@@ -101,4 +106,14 @@ public boolean hasCapability(String capability) {
       return false;
     }
   }
+
+  public static void setRetryPolicy(OzoneClientConfig config) {
+    retryPolicy =
+        HddsClientUtils.createRetryPolicy(config.getMaxReadRetryCount(),
+            TimeUnit.SECONDS.toMillis(config.getReadRetryInterval()));
```

Comment:
Storing this as `static` seems unsafe.

1. Code is spread in 3 places:
- this class
- class that calls `setRetryPolicy`
- class that uses the policy
It is hard to see if any usage may get outdated or `null` value.

2. Concurrent clients may get unexpected results.